crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -983,3 +1063,17 @@ fn find_and_check_object_member(
         _ => None,
     }
 }
+
+/// Traverses up the syntax tree from the given node to find `JsFormalParameter`.
+///
+/// This function traverses up the syntax tree from the given `SyntaxNode` to find the nearest
+/// `JsFormalParameter`. It returns `Some(JsFormalParameter)` if a `JsFormalParameter` is found,
+/// otherwise it returns `None`.
+fn find_js_format_parameter(node: &SyntaxNode<JsLanguage>) -> Option<JsFormalParameter> {
+    node.ancestors()
+        .skip(1)
+        .find_map(|ancestor| match ancestor.kind() {
+            JsSyntaxKind::JS_FORMAL_PARAMETER => ancestor.cast::<JsFormalParameter>(),
+            _ => None,
+        })
```

Comment:
```suggestion
        .find_map(JsFormalParameter::cast)
```

Way simpler. Please check locally if that works, it's possible you need to use `cast_ref` instead.

Reply:
thanks! updated it!